cli/flox-rust-sdk/src/models/environment_ref.rs

```diff
@@ -68,11 +70,11 @@ impl FromStr for EnvironmentRef {
 
 #[derive(Error, Debug)]
 pub enum EnvironmentRefError {
-    #[error("Name format is invalid")]
-    InvalidName,
+    #[error("Name '{0}' is invalid.\nEnvironment names may only contain alphanumeric characters, '.', '_', and '-'.")]
+    InvalidName(String),
 
-    #[error("Name format is invalid")]
-    InvalidOwner,
+    #[error("Owner '{0}' is invalid.\nEnvironment owners may only contain alphanumeric characters, '.', '_', and '-'.")]
+    InvalidOwner(String),
```

Comment:
Discussed sync:
We might want to revisit the pretty formatting with newlines here, since it's inconsistent with other error messages that aim at being printed in an error chain, with a pretty printer being applied externally.

Reply:
I do prefer the simplicity of having the error here rather than in some formatter elsewhere

Reply:
More like, have the error print what's actually wrong with the parsed string and wrap that with context separately, but that can wait till we actually do have more rules to check.